mmv1/third_party/terraform/website/docs/r/compute_firewall_policy_rule.html.markdown

```diff
@@ -0,0 +1,162 @@
+#
+#     This file is managed by Magic Modules (https:#github.com/GoogleCloudPlatform/magic-modules)
+#     and is based on the DCL (https:#github.com/GoogleCloudPlatform/declarative-resource-client-library).
+#     Changes will need to be made to the DCL or Magic Modules instead of here.
+#
+#     We are not currently able to accept contributions to this file. If changes
+#     are required, please file an issue at https:#github.com/hashicorp/terraform-provider-google/issues/new/choose
+#
+# ----------------------------------------------------------------------------
+subcategory: "Compute"
+layout: "google"
+page_title: "Google: google_compute_firewall_policy_rule"
+sidebar_current: "docs-google-compute-firewall-policy-rule"
+description: |-
+  Specific rules to add to a hierarchical firewall policy
+---
+
+# google\_compute\_firewall\_policy\_rule
+
+Hierarchical firewall policy rules let you create and enforce a consistent firewall policy across your organization. Rules can explicitly allow or deny connections or delegate evaluation to lower level policies.
+
+For more information see the [official documentation](https://cloud.google.com/vpc/docs/using-firewall-policies#create-rules)
+
+## Example Usage
+
+```hcl
+resource "google_compute_firewall_policy" "default" {
+  parent      = "organizations/12345"
+  short_name  = "my-policy"
+  description = "Example Resource"
+}
+
+resource "google_compute_firewall_policy_rule" "default" {
+  firewall_policy = google_compute_firewall_policy.default.name
+  description = "Example Resource"
+  priority = 9000
+  enable_logging = true
+  action = "allow"
+  direction = "EGRESS"
+  disabled = false
+  match {
+    layer4_configs {
+      ip_protocol = "tcp"
+      ports = [80, 8080]
+    }
+    dest_ip_ranges = ["11.100.0.1/32"]
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `action` -
+  (Required)
+  The Action to perform when the client connection triggers the rule. Can currently be either "allow" or "deny()" where valid values for status are 403, 404, and 502.
+  
+* `direction` -
+  (Required)
+  The direction in which this rule applies. Possible values: INGRESS, EGRESS
+  
+* `firewall_policy` -
+  (Required)
+  The firewall policy of the resource.
+  
+* `match` -
+  (Required)
+  A match condition that incoming traffic is evaluated against. If it evaluates to true, the corresponding 'action' is enforced.
+  
+* `priority` -
+  (Required)
+  An integer indicating the priority of a rule in the list. The priority must be a positive value between 0 and 2147483647. Rules are evaluated from highest to lowest priority where 0 is the highest priority and 2147483647 is the lowest prority.
+  
+
+
+The `match` block supports:
+    
+* `dest_ip_ranges` -
+  (Optional)
+  CIDR IP address range. Maximum number of destination CIDR IP ranges allowed is 256.
+    
+* `layer4_configs` -
+  (Required)
```

Comment:
Can you file a generator bug that we should order subfields correctly by required > optional?